### ISSUE TITLE
Error Failed to apply {created} template in file header as file info is unavailable when running in github action

### ISSUE THREAD

**Issue by Dragna** [nicklockwood/SwiftFormat#1300]
Hi, 

i have a github action where I run the linux version of swift format, when I make it run I get this error
`error: Failed to apply {created} template in file header as file info is unavailable.`

Here is the command I am running : `./swiftformat_linux . --config ".swiftformat"`

here is my swiftformat config : 

```
# options
--exclude **/Pods # exclude the Pods folder from the formating

# swift version
--swiftversion 5.5

# disabled rules
--disable all
--enable fileHeader

# format options
--header "//\n// Copyright © {created.year} MyCompany.\n//"
```

everything works fine when I'm running swiftformat on my mac machine.

If anyone has any idea on what is going on and what I can do to make it work, that would be super helpful 🙏.

Have a great day 🙂.

**Comment by Dragna** [comment]
it seems that if I use stdin with a stdinpath for a given file it is working, but I don't understand why I should use stdin in this use case . Furthermore what I would like to do is run swiftformat on all swift files and check if a file is modified to block the Pull request, but using stdinpath would mean invoking swiftformat individually for each file which is not really efficient right ?

**Comment by nicklockwood** [comment]
Historically, Swift didn't support the methods that SwiftFormat uses to get the created date on Linux, so the feature is disabled. I'm not sure why it works with stdin.

It looks like it should be supported now though anyway, so I'll enable the feature on Linux too in the next release.

**Comment by Dragna** [comment]
oh great news, thank you for your help 🙏

**Comment by nicklockwood** [comment]
Hmm, I've removed the special case logic for Linux, but looking at the code it seems creation date should have worked anyway, so I'm not sure if the issue you're having is related.

**Comment by Dragna** [comment]
to add more detail swiftformat seems to be running on each files like expected, but i showing the error for each one of the files.

I can run some more tests to help if you tell me what you want me to try 🙂.

**Comment by nicklockwood** [comment]
If you can try building from the develop branch and seeing if that's fixed it, that would be helpful. If it doesn't I'll see if I can reproduce it myself, thanks!

**Comment by Dragna** [comment]
Hi, 
Do you have a way to build the develop branch for Linux ? it doesn’t seem like I can launch the action to do it myself 🙂

**Comment by nicklockwood** [comment]
Sorry, try this: https://github.com/nicklockwood/SwiftFormat/suites/9151577780/artifacts/425777478

**Comment by Dragna** [comment]
Hi, 

We tried running this using the swiftformat docker image (to understand if it was coming from the GHA environment or not) and we have the same problem when we do have the same problem as with GHA 😕.

I tried understanding what could be happening but no clue as of now 🙂

**Comment by nicklockwood** [comment]
Yeah, unfortunately I also have no idea what's happening here. I can't understand why file creation info would be available in one context but not another, since it's evidently finding the file itself.

**Comment by nicklockwood** [comment]
@Dragna the `{created}` template was updated to use git info - does it now work for you or are you still seeing this issue?
